src/memote/biomass.jl

```diff
@@ -0,0 +1,174 @@
+#=
+This file contains a collection of tests based on Memote. See Lieven, C., Beber,
+M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
+testing. Nat Biotechnol 38, 272–276 (2020).
+https://doi.org/10.1038/s41587-020-0446-y for details.
+=#
+
+"""
+$(TYPEDSIGNATURES)
+
+Identify all the biomass reactions in the `model` using both sbo annotations, as
+well biomass strings typically contained in their reaction IDs. Use
+`config.biomass.biomass_strings` to update the list of strings to look for.
+"""
+model_biomass_reactions(model; config = memote_config) = Set(
+    [
+        [rid for rid in reactions(model) if is_biomass_reaction(model, rid)]
+        find_biomass_reaction_ids(model; biomass_strings = config.biomass.biomass_strings)
+    ],
+)
+
+"""
+$(TYPEDSIGNATURES)
+
+Check if model has an ATP maintenance reaction built in (also called a
+non-growth associated maintenance cost). Looks for reaction annotations
+corresponding to the sbo maintenance term, or looks for reaction ids that
+contain the strings listed in `config.biomass.atpm_strings`.
+"""
+model_has_atpm_reaction(model; config = memote_config) =
+    any(is_atp_maintenance_reaction(model, rid) for rid in reactions(model)) ||
+    any(any(occursin.(config.biomass.atpm_strings, Ref(rid))) for rid in reactions(model))
+
+"""
+$(TYPEDSIGNATURES)
+
+Check if the biomass reaction consumes ATP and H₂O, and produces ADP, HO₄P, and
+H⁺. Each of these metabolites have a lookup table mapping them to the name space
+of the model, defined in `config.biomass.growth_metabolites`. These need to be
+set if you use anything other than the BiGG namespace.
+"""
+function atp_present_in_biomass(model; config = memote_config)
+    biomass_rxns = model_biomass_reactions(model; config)
+    x = Dict(biomass_rxns .=> true)
+    for rid in biomass_rxns
+        d = reaction_stoichiometry(model, rid)
+        for (m1, m2) in config.biomass.growth_metabolites
+            if m1 in ["atp", "h2o"] # must consume
+                !haskey(d, m2) && !(d[m2] < 0) && (x[rid] = false)
+            else # must produce
+                !haskey(d, m2) && !(d[m2] > 0) && (x[rid] = false)
```

Comment:
just a note, to reduce the amount of condition combinations in the code, this could be better written more clearly as something like:
```
x[rid] &= get(d, m2, Inf) <= 0
```

Reply:
this is great, thanks!